src/Compilers/VisualBasic/Portable/Operations/VisualBasicOperationFactory_Methods.vb

```diff
@@ -242,46 +242,93 @@ Namespace Microsoft.CodeAnalysis.Operations
                             initializerSyntax = initializerValue.Syntax
                             isImplicit = True
                         End If
                         initializer = OperationFactory.CreateVariableInitializer(initializerSyntax, initializerValue, _semanticModel, isImplicit)
                     End If
-                    builder.Add(OperationFactory.CreateVariableDeclaration(declaration.LocalSymbol, initializer, _semanticModel, declaration.Syntax))
-                ElseIf base.Kind = BoundKind.AsNewLocalDeclarations Then
-                    Dim asNewDeclarations = DirectCast(base, BoundAsNewLocalDeclarations)
-                    Dim localSymbols = asNewDeclarations.LocalDeclarations.SelectAsArray(Of ILocalSymbol)(Function(declaration) declaration.LocalSymbol)
+                Else
+                    Dim asNewDeclarations = DirectCast(first, BoundAsNewLocalDeclarations)
+                    singleDeclarations = asNewDeclarations.LocalDeclarations.SelectAsArray(AddressOf GetVariableDeclarator)
                     Dim initializerSyntax As AsClauseSyntax = DirectCast(asNewDeclarations.Syntax, VariableDeclaratorSyntax).AsClause
                     Dim initializerValue As IOperation = Create(asNewDeclarations.Initializer)
-                    Dim initializer As IVariableInitializerOperation = OperationFactory.CreateVariableInitializer(initializerSyntax, initializerValue, _semanticModel, isImplicit:=False)
-                    builder.Add(OperationFactory.CreateVariableDeclaration(localSymbols, initializer, _semanticModel, asNewDeclarations.Syntax))
+                    initializer = OperationFactory.CreateVariableInitializer(initializerSyntax, initializerValue, _semanticModel, isImplicit:=False)
                 End If
+
+                builder.Add(New VariableDeclaration(singleDeclarations,
+                                                         initializer,
+                                                         _semanticModel,
+                                                         declarationGroup.Key,
+                                                         type:=Nothing,
+                                                         constantValue:=Nothing,
+                                                         isImplicit:=False))
             Next
 
             Return builder.ToImmutableAndFree()
         End Function
 
-        Private Function GetUsingStatementDeclaration(resourceList As ImmutableArray(Of BoundLocalDeclarationBase), syntax As SyntaxNode) As IVariableDeclarationsOperation
-            Return New VariableDeclarationStatement(
+        Private Function GetVariableDeclarator(boundLocalDeclaration As BoundLocalDeclaration) As IVariableDeclaratorOperation
+            Dim initializer As Lazy(Of IVariableInitializerOperation) = New Lazy(Of IVariableInitializerOperation)(
```

Comment:
do we want to put this in the cache? as long as it never shows up as root or top level element such as statement, won't matter much but just asking.

Reply:
This really won't work well in the cache, since we already have create methods for the children of BoundLocalDeclarationBase.

---
In reply to: [148097014](https://github.com/dotnet/roslyn/pull/22842#discussion_r148097014) [](ancestors = 148097014)